### PR TITLE
CU/7056-Dylan-LettersUnitTests

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Description of Change
letters screen unit tests

## Screenshots/Video
N/A

## Testing
yarn test

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
L src/screens/BenefitsScreen/Letters/BenefitSummaryServiceVerification/BenefitSummaryServiceVerification.test.tsx
S src/screens/BenefitsScreen/Letters/NoLettersScreen/NoLettersScreen.test.tsx
S src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx
S src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.test.tsx

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

**Review by theodur — CHANGES_REQUESTED** [review]
Same for this PR
https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458

**Comment by Sparowhawk** [comment]
> Same for this PR [#7148 (review)](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7148#pullrequestreview-1706588458)

updated

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx [hunk 1]
```diff
@@ -1,9 +1,8 @@
 import 'react-native'
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:1]: Another one that can be removed

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx [hunk 2]
```diff
@@ -29,24 +26,29 @@ context('LettersOverviewScreen', () => {
...
+    expect(screen.getByText('Downloaded documents will list your address as:')).toBeTruthy()
+    expect(screen.getByTestId('Mailing address Add your mailing address')).toBeTruthy()
+    expect(screen.getByText('If this address is incorrect you may want to update it, but your letter will still be valid even with the incorrect address.')).toBeTruthy()
+    expect(screen.getByRole('button', { name: 'Review letters' })).toBeTruthy()
   })
 
-  it('should go to edit address when the address is pressed', async () => {
-    testInstance.findAllByType(Pressable)[0].props.onPress()
+  it('should go to edit address when the address is pressed', () => {
+    fireEvent.press(screen.getByTestId('Mailing address Add your mailing address'))
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:47]:
```suggestion
    fireEvent.press(screen.getByRole('button', { name: 'Mailing address Add your mailing address' }))
```

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx [hunk 3]
```diff
@@ -29,24 +26,29 @@ context('LettersOverviewScreen', () => {
...
 
   beforeEach(() => {
     initializeTestInstance()
   })
 
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+  it('initializes correctly', () => {
+    expect(screen.getByText('Downloaded documents will list your address as:')).toBeTruthy()
+    expect(screen.getByTestId('Mailing address Add your mailing address')).toBeTruthy()
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:41]:
```suggestion
    expect(screen.getByRole('button', { name: 'Mailing address Add your mailing address' })).toBeTruthy()
```

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx [hunk 4]
```diff
@@ -1,9 +1,8 @@
...
-// Note: test renderer must be required after react-native.
-import { ReactTestInstance } from 'react-test-renderer'
-import { context, mockNavProps, render, RenderAPI } from 'testUtils'
-import { Pressable } from 'react-native'
+import { screen, fireEvent } from '@testing-library/react-native'
+
+import { context, mockNavProps, render } from 'testUtils'
 import { LettersOverviewScreen } from './index'
 import { profileAddressOptions } from 'screens/HomeScreen/ProfileScreen/ContactInformationScreen/AddressSummary'
 import { when } from 'jest-when'
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:8]: This import can be grouped with the others at the top

File: VAMobile/src/screens/BenefitsScreen/Letters/NoLettersScreen/NoLettersScreen.test.tsx [hunk 5]
```diff
@@ -1,21 +1,16 @@
 import 'react-native'
```
Comment (theodur) [inline VAMobile/src/screens/BenefitsScreen/Letters/NoLettersScreen/NoLettersScreen.test.tsx:1]: Can remove this also

**Review by theodur — CHANGES_REQUESTED** [review]

File: VAMobile/src/screens/BenefitsScreen/Letters/BenefitSummaryServiceVerification/BenefitSummaryServiceVerification.test.tsx [hunk 6]
```diff
@@ -1,18 +1,13 @@
 import 'react-native'
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/BenefitSummaryServiceVerification/BenefitSummaryServiceVerification.test.tsx:1]: removed

File: VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.test.tsx [hunk 7]
```diff
@@ -1,13 +1,10 @@
 import 'react-native'
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.test.tsx:1]: removed

File: VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.test.tsx [hunk 8]
```diff
@@ -24,66 +21,60 @@ jest.mock('store/slices', () => {
...
   beforeEach(() => {
     initializeTestInstance()
   })
 
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+  it('initializes correctly', () => {
+    expect(screen.getByText('Review letters')).toBeTruthy()
+    expect(screen.getByTestId('Letters: header-page')).toBeTruthy()
+    expect(screen.getByText('Review letter')).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.test.tsx:44]: updated

File: VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.test.tsx [hunk 9]
```diff
@@ -24,66 +21,60 @@ jest.mock('store/slices', () => {
...
   }
 
   beforeEach(() => {
     initializeTestInstance()
   })
 
-  it('initializes correctly', async () => {
-    expect(component).toBeTruthy()
+  it('initializes correctly', () => {
+    expect(screen.getByText('Review letters')).toBeTruthy()
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/GenericLetter/GenericLetter.test.tsx:42]: there is no duplicate check for this one it is slightly different as its the screen title

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersListScreen.test.tsx [hunk 10]
```diff
@@ -1,7 +1,7 @@
 import 'react-native'
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/LettersListScreen.test.tsx:1]: removed

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx (hunk 1 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:1]: removed

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx (hunk 4 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:8]: updated

File: VAMobile/src/screens/BenefitsScreen/Letters/NoLettersScreen/NoLettersScreen.test.tsx (hunk 5 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/NoLettersScreen/NoLettersScreen.test.tsx:1]: removed

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/LettersOverviewScreen.test.tsx:47]: updated

**Review by alexandec — DISMISSED** [review]

File: VAMobile/src/screens/BenefitsScreen/Letters/LettersListScreen.test.tsx
[the remark this answers is not in the document]
Comment (theodur) [reply VAMobile/src/screens/BenefitsScreen/Letters/LettersListScreen.test.tsx:1]: I noticed the async calls in the first two tests are still there

[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/BenefitsScreen/Letters/LettersListScreen.test.tsx:1]: fixed

**Review by theodur — APPROVED** [review]

**Review by alexandec — APPROVED** [review]

**Pull request merged**